controllers/datafoundation.go

```diff
@@ -171,6 +181,21 @@ func (r *dataFoundationReconciler) reconcilePhases() (ctrl.Result, error) {
 		}
 
 	} else {
+
+		missingCRDs := []string{}
+		// Checking for CRDs that are installed after the agent was started and add watchers for the new CRDs
+		if !r.ManagedFusionOfferingReconciler.AvailableCRDs[StorageClusterCRDName] && len(missingCRDs) < 1 {
+			missingCRDs = append(missingCRDs, StorageClusterCRDName)
+		}
+		if !r.ManagedFusionOfferingReconciler.AvailableCRDs[OCSInitializationCRDName] && len(missingCRDs) < 2 {
+			missingCRDs = append(missingCRDs, OCSInitializationCRDName)
+		}
+		if len(missingCRDs) != 0 {
+			if err := utils.HandleMissingWatchers(r.Client, missingCRDs); err != nil {
+				return ctrl.Result{}, err
+			}
```

Comment:
@rohan47 This seems like a regression of the older revision. Where are the get operations? why are you keeping a list? why do we have only a single call to `HandleMissingWatchers`?

I would expect to see a simple get+HandleMissingWatchers call, for each of the CRD in question. The simplest solution possible.